--- /usr/share/condor-ce/config.d/01-ce-router-defaults.conf
+++ /usr/share/condor-ce/config.d/01-ce-router-defaults.conf
@@ -255,19 +255,19 @@ JOB_ROUTER_TRANSFORM_Queue @=jrt
 
 JOB_ROUTER_TRANSFORM_BatchRuntime @=jrt
     # 'BatchRuntime' is in seconds but admins configure 'default_maxWallTime' and 'ROUTED_JOB_MAX_TIME' and remote
-    # submitters set 'maxWallTime' in minutes. Remote submitters set 'BatchRuntime' in minutes
+    # submitters set 'maxWallTime' in minutes. Remote submitters set 'BatchRuntime' in seconds.
     # Set the remote batch runtime used by non-HTCondor batch systems to one of the following, in order:
     # 1. 'maxWalltime' (minutes) from the source job
     # 2. 'BatchRuntime' (seconds) from the source job
     # 3. 'default_maxWallTime' (minutes) from the route
     # 4. 'ROUTED_JOB_MAX_TIME' (minutes) from the config
     if defined MY.maxWallTime
-        SET BatchRuntime 60*maxWallTime
+        EVALSET BatchRuntime 60*maxWallTime
     elif defined MY.orig_BatchRuntime
         # do nothing, BatchRuntime already set
     elif defined default_maxWallTime
         # default_maxWallTime can be an attribute reference here
-        SET BatchRuntime 60*$(default_maxWallTime)
+        EVALSET BatchRuntime 60*$(default_maxWallTime)
     else
         EVALSET BatchRuntime 60*$(ROUTED_JOB_MAX_TIME:4320)
     endif
--- /usr/share/condor-ce/condor_ce_router_defaults
+++ /usr/share/condor-ce/condor_ce_router_defaults
@@ -132,13 +132,13 @@ JOB_ROUTER_DEFAULTS_GENERATED @=jrd
 
     /* BatchRuntime is in seconds but users configure default_maxWallTime and ROUTED_JOB_MAX_TIME in minutes */
     copy_BatchRuntime = "orig_BatchRuntime";
-    set_BatchRuntime = ifThenElse(maxWallTime isnt undefined,
-                                  60*maxWallTime,
-                                  ifThenElse(orig_BatchRuntime isnt undefined,
-                                             orig_BatchRuntime,
-                                             ifThenElse(default_maxWallTime isnt undefined,
-                                                        60*default_maxWallTime,
-                                                        60*$(ROUTED_JOB_MAX_TIME))));
+    eval_set_BatchRuntime = ifThenElse(maxWallTime isnt undefined,
+                                       60*maxWallTime,
+                                       ifThenElse(orig_BatchRuntime isnt undefined,
+                                                  orig_BatchRuntime,
+                                                  ifThenElse(default_maxWallTime isnt undefined,
+                                                             60*default_maxWallTime,
+                                                             60*$(ROUTED_JOB_MAX_TIME))));
 
     set_CondorCE = 1;
     eval_set_CERequirements = ifThenElse(default_CERequirements isnt undefined,
